src/shared_modules/keystore/src/keyStore.cpp

```diff
@@ -0,0 +1,128 @@
+#include "keyStore.hpp"
+#include "evpHelper.hpp"
+#include "loggerHelper.h"
+#include "rocksDBWrapper.hpp"
+#include "rsaHelper.hpp"
+
+// Database constants, based on the keystore path.
+constexpr auto DATABASE_PATH {"queue/keystore"};
+
+// File constants, used in the version 1 of the keystore, currently in use only for the upgrade to version 2, that uses
+// AES 256 encryption, without the need of the private key file.
+constexpr auto PRIVATE_KEY_FILE {"etc/sslmanager.key"};
+
+// Keystore constants, KS_NAME is used in the logs as tag.
+// KS_VERSION is the current version of the keystore. Used to identify the version of the keystore in the database.
+// KS_VERSION_FIELD is the field used to store the version of the keystore in the database.
+constexpr auto KS_NAME {"keystore"};
+constexpr auto KS_VERSION {"2"};
+constexpr auto KS_VERSION_FIELD {"version"};
+
+static void upgrade(Utils::RocksDBWrapper& keystoreDB, const std::string& columnFamily)
+{
+    std::string versionValue;
+    std::string rawValue;
+
+    // If the version field does not exist, it means that the keystore has not been upgraded yet.
+    if (!keystoreDB.get(KS_VERSION_FIELD, versionValue, columnFamily))
+    {
+        try
+        {
+            // Check if the private key file exists
+            if (!std::filesystem::exists(PRIVATE_KEY_FILE))
+            {
+                throw std::runtime_error("Private key file not found.");
+            }
+
+            // Upgrade all keys
```

Comment:
Nitpick: rawValue can have a smaller scope
```suggestion

    // If the version field does not exist, it means that the keystore has not been upgraded yet.
    if (!keystoreDB.get(KS_VERSION_FIELD, versionValue, columnFamily))
    {
        try
        {
            // Check if the private key file exists
            if (!std::filesystem::exists(PRIVATE_KEY_FILE))
            {
                throw std::runtime_error("Private key file not found.");
            }

            std::string rawValue;
            // Upgrade all keys
```